src/main/java/com/contrastsecurity/maven/plugin/sdkx/DefaultContrastScanSDK.java

```diff
@@ -0,0 +1,209 @@
+    final String url =
+        String.join(
+            "/",
+            restURL,
+            "sast",
+            "organizations",
+            organizationId,
+            "projects",
+            request.getProjectId(),
+            "code-artifacts");
+    final String boundary = "ContrastFormBoundary" + ThreadLocalRandom.current().nextLong();
+    final String header =
+        "--"
+            + boundary
+            + CRLF
+            + "Content-Disposition: form-data; name=\"filename\"; filename=\""
+            + file.getFileName().toString()
+            + '"'
+            + CRLF
+            + "Content-Type: "
+            + determineMime(file)
+            + CRLF
+            + "Content-Transfer-Encoding: binary"
+            + CRLF
+            + CRLF;
+    final String footer = CRLF + "--" + boundary + "--" + CRLF;
+    final long contentLength = header.length() + Files.size(file) + footer.length();
+
+    final HttpURLConnection connection = contrast.makeConnection(url, "POST");
+    connection.setDoOutput(true);
+    connection.setDoInput(true);
+    connection.setRequestProperty("Content-Type", "multipart/form-data; boundary=" + boundary);
+    connection.setFixedLengthStreamingMode(contentLength);
+    try (OutputStream os = connection.getOutputStream();
+        PrintWriter writer =
+            new PrintWriter(new OutputStreamWriter(os, StandardCharsets.US_ASCII), true)) {
+      writer.append(header).flush();
+      Files.copy(file, os);
+      os.flush();
+      writer.append(footer).flush();
+    }
+    final int rc = connection.getResponseCode();
+    // for consistency with other SDK methods, throw UnauthorizedException when request's
+    // authentication is rejected. Unlike other SDK requests, do not conflate all 400 errors with an
+    // authentication problem
+    if (rc == HttpURLConnection.HTTP_FORBIDDEN || rc == HttpURLConnection.HTTP_UNAUTHORIZED) {
+      throw new UnauthorizedException(rc);
+    }
+    if (rc != 201) {
+      throw new ContrastAPIException(rc, "Failed to upload code artifact to Contrast Scan");
+    }
+    try (Reader reader = new InputStreamReader(connection.getInputStream())) {
+      return gson.fromJson(reader, CodeArtifact.class);
+    }
+  }
+
+  @Override
+  public Scan startScan(final String organizationId, final StartScanRequest request)
+      throws IOException, UnauthorizedException {
+    // requests made with ContrastSDK.makeRequest must have their path prepended with "/"
+    final String path =
+        String.join(
+            "/",
+            "",
+            "sast",
+            "organizations",
+            organizationId,
+            "projects",
+            request.getProjectId(),
+            "scans");
+    final String json = gson.toJson(request);
+    try (Reader reader =
+        new InputStreamReader(
+            contrast.makeRequestWithBody(HttpMethod.POST, path, json, MediaType.JSON))) {
+      return gson.fromJson(reader, Scan.class);
```

Comment:
Good catch, but this behavior is consistent with the rest of the Contrast Java SDK. I'd prefer to address this with the whole SDK at once, so can we kick this can down the road so that we're at least consistent for now?